pkg/security/probe/sysctl/snapshot.go

```diff
@@ -247,6 +257,75 @@ func (s *Snapshot) snapshotKernelCmdline(ignoredBaseNames []string) error {
 	return nil
 }
 
+func (s *Snapshot) getKernelConfigPath() (string, error) {
+	kernelVersion, err := os.ReadFile(kernel.HostProc("/sys/kernel/osrelease"))
+	if err != nil {
+		return "", err
+	}
+	configPath := fmt.Sprintf(kernel.HostBoot("/config-%s"), strings.TrimSpace(string(kernelVersion)))
+	if _, err := os.Stat(configPath); err == nil {
+		return configPath, nil
+	}
+	procConfigGZ := kernel.HostProc("/config.gz")
+	if _, err := os.Stat(procConfigGZ); err == nil {
+		return procConfigGZ, nil
+	}
+	return "", fmt.Errorf("kernel config not found")
+}
+
+func (s *Snapshot) parseKernelConfig(r io.Reader, kernelCompilationFLags []string) error {
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if strings.HasPrefix(line, "#") {
+			if strings.HasSuffix(line, "is not set") {
+				key := strings.Fields(line)[1]
+				if slices.Contains(kernelCompilationFLags, key) {
```

Comment:
`bytes.Fields` tends to be better from an allocation point of view, if you only need one/few of the fields.
```suggestion
				key := bytes.Fields(line)[1]
				if slices.Contains(kernelCompilationFlags, string(key)) {
```

Reply:
Updated, thanks !